### PR TITLE
Http4s 1.0.0-M*

### PR DESCRIPTION
**Pull request by zarthross**
I'll set this to a `series/something` branch later, wanted to get this out quick so others know this is being worked on.

(Removed HAL support, if someone wants it back we can revert 81e2dff4aa48d21da27b1f9c6a0e1488b356f038 and clean it up.)

### CODE REVIEW

**Comment by balthz** [comment]
Is this tracking progress to get Rho to CE3? Or does the main branch already support CE3?

File: swagger/src/test/scala/org/http4s/rho/swagger/SwaggerSupportSpec.scala [hunk 1]
```diff
@@ -239,7 +239,7 @@ class SwaggerSupportSpec extends Specification {
     "Check metadata in API listing" in {
       val service = metaDataRoutes.toRoutes(createRhoMiddleware(swaggerRoutesInSwagger = true))
 
-      val r = Request[IO](GET, Uri(path = "/swagger.json"))
+      val r = Request[IO](GET, Uri(path = Uri.Path.fromString("/swagger.json")))
```
Comment (rossabaker) [inline swagger/src/test/scala/org/http4s/rho/swagger/SwaggerSupportSpec.scala:242]: I think these are going to need to be `unsafeFromString` in M20.  You can also consider the Uri interpolator: `uri"/swagger.json"`.

**Review by rossabaker — COMMENTED** [review]
Sorry, this apparently got swept up in one of my notification bankruptcies.  I think it's on the right track.  A lot of this is going to be compiler driven.

**Comment by zarthross** [comment]
> Is this tracking progress to get Rho to CE3? Or does the main branch already support CE3?

Yes, I'll be putting some time in hopefully this weekend and try to create the same kind of split support Http4s has between CE2 and CE3.

**Comment by balthz** [comment]
> > Is this tracking progress to get Rho to CE3? Or does the main branch already support CE3?
> 
> Yes, I'll be putting some time in hopefully this weekend and try to create the same kind of split support Http4s has between CE2 and CE3.

I’m unfortunately not acquainted with Rho’s internals but it’s the only project in my project’s Cats ecosystem still on CE2. Is there any way I can help here? For example, would it be sufficient to _only_ support CE3 going forward? That might make things easier for me or others to help the move.

**Comment by zarthross** [comment]
Only supporting CE3 might make this repo easier to maintain, but I probably won't be able to migrate my work projects to CE3 for a while.  In either case, its important to note that Http4s is on 1.0.0-M21 for its CE3 support, and so we won't be releasing anything stable CE3 for a while.  I'm going to try and get a M1 released once i sort out the header changes.  I'll have a PR over to Http4s for some fixes over this next weekend.

Sorry this is taking a while.

**Comment by rossabaker** [comment]
Scala 3 final is coming, and we want to make sure it doesn't throw us one last curveball.  The goal is to release 0.22 (for CE2) and something to be named later for CE3 (1.0 if we have a good few weeks, or an 0.23 forked off it if we're not ready), to have *something* stable for integrations to depend on.  Expect those ASAP after Scala 3's official release.

**Comment by zarthross** [comment]
Will open a new PR on a new branch for this shortly.

**Comment by zarthross** [comment]
Replaced with https://github.com/http4s/rho/pull/549

**Pull request closed without merging**